smallrye-llm-langchain4j-portable-extension/src/main/java/io/smallrye/llm/core/langchain4j/portableextension/LangChain4JAIServicePortableExtension.java

```diff
@@ -44,37 +50,54 @@ <T> void processAnnotatedType(@Observes @WithAnnotations({ RegisterAIService.cla
      */
     void processInjectionPoints(@Observes ProcessInjectionPoint<?, ?> event) {
         if (event.getInjectionPoint().getBean() == null) {
-            processInjectionPoint(event);
+            componentInjectionPoints.add(event.getInjectionPoint());
         }
 
         if (Instance.class.equals(Reflections.getRawType(event.getInjectionPoint().getType()))) {
```

Comment:
This line should be removed based on the issue conversation #27

Reply:
This PR is a complete rewrite, and thus, this method was removed. It just registers the InjectionPoint so that the Bean can be registered on afterBeanDiscovery phase .

Reply:
What I tried to explain [here](https://github.com/smallrye/smallrye-llm/issues/27#issuecomment-2324674098) is that
 `event.getInjectionPoint().getType()` should never give you `Instance<X>`. According to the specification, it should return just `X` which means that this `if` condition you still have here will not trigger.

Reply:
Hmmm, interesting observation. I guess the first if statement might work for both non-Instance, as well as Instance fields. 🤔

Reply:
Sorry, came back from a long task for my customers... Well, could we try to show the problem in the unit test ? Adding a specific test with "@Inject
    Instance<MyDummyAIService> myDummyAIServiceInstance;" inside ?

Reply:
> Sorry, came back from a long task for my customers... Well, could we try to show the problem in the unit test ? Adding a specific test with "@Inject Instance < MyDummyAIService > myDummyAIServiceInstance;" inside ? I could try

I'm working on it.